src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -155,7 +162,11 @@ public void run() {
      * Returns the file that records the last/current polling activity.
      */
     public File getLogFile() {
-        return new File(job.getRootDir(), "github-polling.log");
+        if (job == null) {
+            return null;
+        } else {
```

Comment:
Not sure about coding style, but I wouldn't use an `else` here ("else after return").

Reply:
If we want to get fancy, we can also update the signature to be a `RandomAccessFile`: https://gist.github.com/er1c/bcec81835c4eea05a208995e7c37a4da to ensure it gets deleted after creation, and guarantee it isn't null.  That requires a signature change tho.

Reply:
I updated this block of code, and added additional null checks/`@CheckForNull` annotation.